### PR TITLE
Add a test to read and write GCS with gRPC enabled.

### PR DESCRIPTION
**Pull request by shunping**
The gRPC feature is an experimental feature from GCS. As it is approaching GA, we need to add some basic test to make sure it works with Beam.

The test will be invoked from a post commit test github action: https://github.com/apache/beam/blob/e6cc0809646da44b5621b3772704d645b79d8253/.github/workflows/beam_PostCommit_Java.yml#L80

`:JavePostCommit` (defined in `build.gradle.tks`)
-> `:sdks:java:extensions:google-cloud-platform-core:postCommit` (defined in `sdks/java/extensions/google-cloud-platform-core/build.gradle`)
-> `:sdks:java:extensions:google-cloud-platform-core:integrationTestKms`
-> `sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java`

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by shunping** [comment]
assign set of reviewers

**Comment by shunping** [comment]
assign set of reviewers

**Comment by Abacn** [comment]
Could you please trigger Java PostCommit (add an empty file `.github/trigger_files/beam_PostCommit_Java.json`)

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java [hunk 1]
```diff
@@ -76,4 +86,67 @@ public void testRewriteMultiPart() throws IOException {
...
+    // whether we can remove the assertion.
+    String tempLocationWithoutGrpc = "gs://temp-storage-for-end-to-end-tests-cmek/temp";
+    String wrongFilename =
+        String.format(
+            "%s/GcsUtilIT-%tF-%<tH-%<tM-%<tS-%<tL.testWriteAndReadGcsWithGrpc.txt",
+            tempLocationWithoutGrpc, new Date());
+    assertThrows(IOException.class, () -> writeGcsTextFile(gcsUtil, wrongFilename, testContent));
+
+    // Write a test file in a bucket with gRPC enabled.
+    String tempLocationWithGrpc = "gs://gcs-grpc-team-apache-beam-testing/temp";
```
Comment (Abacn) [inline sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java:117]: Let's make this configurable. Define this as a static final field, and detect whether a system property (e.g. `gcs_temp_bucket_grpc` exists) which can override the default.

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java [hunk 2]
```diff
@@ -76,4 +86,67 @@ public void testRewriteMultiPart() throws IOException {
...
+    GcsUtil gcsUtil = gcsOptions.getGcsUtil();
+    assertNotNull(gcsUtil);
+
+    String testContent = "This is a test string.";
+
+    // Write a test file in a bucket without gRPC enabled.
+    // This assumes that GCS gRPC feature is not enabled in every bucket by default.
+    // If the following assertion fails, we can revisit the GA status of this feature and check
+    // whether we can remove the assertion.
+    String tempLocationWithoutGrpc = "gs://temp-storage-for-end-to-end-tests-cmek/temp";
```
Comment (Abacn) [inline sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java:109]: Consider use tempRoot obtained from TestPipelineOptions. Then this is configurable.

**Review by Abacn — COMMENTED** [review]
I see tests passed. In general the test should be able to run by arbitrary party, not necessary to access apache-beam-testing project. Thus the real gcp resource should be configurable.

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java (hunk 2 above)
Comment (shunping) [reply sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java]: Done. Please review the latest revision.

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java (hunk 1 above)
Comment (shunping) [reply sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtilIT.java]: Done.

**Comment by shunping** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by shunping** [comment]
Run Java PreCommit

**Comment by shunping** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by shunping** [comment]
gentle ping @Abacn

**Review by Abacn — APPROVED** [review]

**Pull request merged**